app/src/main/java/com/fpvout/digiview/MainActivity.java

```diff
@@ -21,25 +24,36 @@
 import android.view.ViewGroup;
 import android.view.WindowManager;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import androidx.appcompat.app.ActionBar;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import androidx.core.content.res.ResourcesCompat;
 import androidx.preference.PreferenceManager;
 
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+
+import net.ossrs.rtmp.ConnectCheckerRtmp;
+
 import java.util.HashMap;
 
 import io.sentry.SentryLevel;
 import io.sentry.android.core.SentryAndroid;
 
-import static com.fpvout.digiview.VideoReaderExoplayer.VideoZoomedIn;
-
-public class MainActivity extends AppCompatActivity implements UsbDeviceListener {
-    private static final String ACTION_USB_PERMISSION = "com.fpvout.digiview.USB_PERMISSION";
+public class MainActivity extends AppCompatActivity implements UsbDeviceListener, ConnectCheckerRtmp, ActivityCompat.OnRequestPermissionsResultCallback {
```

Comment:
Not sure we should extend the mainActivity with features specific classes...
We can instead create an object of this class when needed (like when we click on the "stream" button) and setup the event listeners at this moment :)